include/pybind11/numpy.h

```diff
@@ -1551,8 +1551,10 @@ struct vectorize_helper {
             "pybind11::vectorize(...) requires a function with at least one vectorizable argument");
 
 public:
-    template <typename T>
-    explicit vectorize_helper(T &&f) : f(std::forward<T>(f)) { }
+    template <typename T,
```

Comment:
A 1-line hint (comment) that this is to not shadow the default copy constructor would be ideal. Maybe with a link to the clang-tidy documentation, for background.